### ISSUE TITLE
Logitech F310 support ?

### ISSUE THREAD

**Issue by Sasounet** [wiseman/webflight-gamepad#6]
I guess I need to beg you for adding support for the Logitech F310. I would not even know how to start modifying the code. As it is, only the left joystick works (for forward, back, left right). The other joystick and all buttons return null events.

I think the F310 is a fairly standard gamepad. Let me know if I can help you in any way.

Looking forward to fly my drone with a real controller!!

Thank you.

**Comment by wiseman** [comment]
Sorry, I don't have a Logitech F310 so it's unlikely I'll be able to add support.  If you get it working I would welcome a pull request though!

**Comment by Sasounet** [comment]
Hi,

Yes I was able to make it work. I was dubious at first that I could even understand the source code because I don't know how to program in javascript but it was very clear, at least the section that reads the controller axis.

I am not familiar with github and I don't know what a pull request is. But here is what I changed in the gamepad.js file:

```
Gamepad.prototype.updateDisplay = function(gamepadId) {
  var gamepad = this.gamepads[gamepadId];
  var roll = gamepad.axes[0];
  var pitch = gamepad.axes[1];
  var yaw = gamepad.axes[2];     <------- This was 5 for your controller and is 2 for the F310
  var altitude = gamepad.axes[3];    <------ This was 6 for your controller and is 3 for the F310
```

However, although the controller was now working correctly it was very difficult (at least for me as I am not an expert) to control the drone because when all joysticks where released the drone did not fall into automatic stabilization mode and was drifting a lot. So I added some lines of code to add this feature. I guess professional droners can fly the ar drone without automatic stabilization but for novice like me it is difficult!

This is the final code section:

```
Gamepad.prototype.updateDisplay = function(gamepadId) {
  var gamepad = this.gamepads[gamepadId];
  var roll = gamepad.axes[0];
  var pitch = gamepad.axes[1];
  var yaw = gamepad.axes[2];
  var altitude = gamepad.axes[3];
  this.sendCommands(pitch, roll, yaw, altitude);
```

//  This conditional statement checks if the all joysticks of the controller are released (in there dead zone) and if so, put the drone in automatic stabilization

```
  if (Math.abs(roll) < 0.1 && Math.abs(pitch) < 0.1 && Math.abs(yaw) < 0.1 && Math.abs(altitude) < 0.1)   
  {
     this.cockpit.socket.emit("/pilot/drone", {action : 'stop' });
  }
```

  };

I let you decide if you want to add this functionality in your code or not.

Eventually I will try to program the controller push buttons to deal with drone function like 'takeoff', 'land', 'flip' etc so that I don't need to reach for the keyboard.

I am very happy with the final result and I thank you for that. Not only I can now control the drone with a nice physical controller but I also get a much better range. Before with my phone or ipad I would get 50 to 150 feet maximum. With the macbook air wifi signal I can go up to 1000 feet!!

Have a nice day.
